### PR TITLE
drivers: espi: Kconfig cleanup for UART mapping

### LINKED ISSUE #45001: drivers: espi: UART to SoC mapping shall use device tree
Currently using KConfig for eSPI virtual UART port to physical SoC UART

Need to use existing device tree selected uart in board device tree

### PR DESCRIPTION
**Pull request by albertofloyd**
Remove KConfig to map SoC physical UART to eSPI virtual UART.

Closes https://github.com/zephyrproject-rtos/zephyr/issues/45001

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

### CODE REVIEW

**Comment by albertofloyd** [comment]
@galak @mbolivar is there a way to get the instance logical id for a node.
I considered (DT_CHOSEN(zephyr_console) but couldn't figure out how to identify the index of the instance selected i.e. 0 for uart0, 1 for uart1. etc

**Comment by mbolivar-nordic** [comment]
> @galak @mbolivar is there a way to get the instance logical id for a node.

Can you rephrase this using the jargon in https://docs.zephyrproject.org/latest/build/dts/intro.html#syntax-and-structure?

Like are you looking for the unit address in a node's name? You should be able to get that out of the reg property or we can add a separate API for it if that won't work for some reason.

If you're looking for "the" node label, then no, you can't do that because a node can have multiple labels.

**Comment by albertofloyd** [comment]
> > @galak @mbolivar is there a way to get the instance logical id for a node.
> 
> Can you rephrase this using the jargon in https://docs.zephyrproject.org/latest/build/dts/intro.html#syntax-and-structure?
> 
> Like are you looking for the unit address in a node's name? You should be able to get that out of the reg property or we can add a separate API for it if that won't work for some reason.
> 
> If you're looking for "the" node label, then no, you can't do that because a node can have multiple labels.

Ideally instead of checking for every UART node status as okay or not as this PR does.  I wanted to be able to obtain first the **uart node chosen for the console**.  https://docs.zephyrproject.org/latest/build/dts/intro.html#aliases-and-chosen-nodes
That works.

But then once I obtain the **node** i don't need any **node property** such as label, address, etc, but I want to identify the node instance for said node. i.e. if choosen node is uart0, I want to get '0', if node uart1 , '1', etc
I don't see any other option other than adding a new property instance but I'm unsure if that's the right way or there is an alternative.

			uart0: uart@400f2400 {
				compatible = "microchip,xec-uart";
                                **instance = 0**;
				reg = <0x400f2400 0x400>;
				interrupts = <40 0>;
				clock-frequency = <1843200>;
				current-speed = <38400>;
				girqs = <15 0>;
				pcrs = <2 1>;
				ldn = <9>;
				status = "disabled";
			};
			uart1: uart@400f2800 {
				compatible = "microchip,xec-uart";
                                **instance = 1**;
				reg = <0x400f2800 0x400>;
				interrupts = <41 0>;
				clock-frequency = <1843200>;
				current-speed = <38400>;
				girqs = <15 1>;
				pcrs = <2 2>;
				ldn = <10>;
				status = "disabled";
			};

**Comment by mbolivar-nordic** [comment]
> if choosen node is uart0, I want to get '0', if node uart1 , '1', etc

I suspect this is an XY problem. What are you really trying to do?

But yes, the short answer is that you're looking for a piece of a node label associated with the node, and no, you can't do that with the current DT API. The slightly longer answer is that it's done that way for a reason because this is usually not what you want.

**Comment by albertofloyd** [comment]
> But then once I obtain the **node** i don't need any **node property** such as label, address, etc, but I want to identify the node instance for said node. i.e. if choosen node is uart0, I want to get '0', if node uart1 , '1', etc
> I don't see any other option other than adding a new property instance but I'm unsure if that's the right way or there is an alternative.

eSPI bus allows to tunnel different traffic over its peripheral channel. 
In this case,  UART traffic coming from eSPI host  (any OS, BIOS, etc) into the eSPI bus, the eSPI target HW intercepts this an redirects it to the physical UART.

eSPI host writes to virtual UART ->  eSPI host HW intercepts and tunnels -> eSPI bus -> eSPI target HW intercepts -> UART 

Currently we set this UART virtual to physical UART mapping based on each board using ESPI_PERIPHERAL_UART_SOC_MAPPING, but it is not scalable enough to handle multiple KConfig, hence wanted to move it to device tree.

Option #1 was to use the chosen UART console, since in all cases it's the same than UART used by eSPI HW., but as you pointed not feasible. 

I guess option #2 has to use a new eSPI device property so each board can set in their dts. 
Let me, re-work the solution in this PR that way.

**Comment by mbolivar-nordic** [comment]
Thanks for the details. I'm not 100% sure without seeing the new property you propose, but it does sound like you could solve this with an additional phandle property that links to the target UART of choice, yes.
